unstructured/ingest/interfaces.py

```diff
@@ -94,8 +100,24 @@ def process_file(self):
         self.isd_elems_no_filename = []
         for elem in isd_elems:
             # type: ignore
-            elem["metadata"].pop("filename", None)  # type: ignore[attr-defined]
-            elem.pop("coordinates")  # type: ignore[attr-defined]
+            if self.config.metadata_exclude is not None:
+                ex_list = self.config.metadata_exclude.split(",")
+                for ex in ex_list:
+                    elem["metadata"].pop(ex, None)  # type: ignore[attr-defined]
+            elif self.config.metadata_include is not None:
+                in_list = self.config.metadata_include.split(",")
+                for k in elem["metadata"]:
+                    if k not in in_list:
+                        elem["metadata"].pop(k, None)  # type: ignore[attr-defined]
+
+            in_list = self.config.fields_include.split(",")
+            ex_list = []
+            for k in elem:
+                if k not in in_list:
+                    ex_list.append(k)
+            for k in ex_list:
+                elem.pop(k)  # type: ignore[attr-defined]
```

Comment:
Instead of iterating the lists twice maybe just a quick comprehension here? Something like: 
```suggestion
            {k: v for k, v in elem.items() if k in in_list}
```

Reply:
Thanks for the feedback, this looks much cleaner for sure.